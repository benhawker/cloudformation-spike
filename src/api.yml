swagger: "2.0"
schemes:
  - https
info:
  title: Product API
  version: 0.0.1
  description: |
    Example API Gateway implementation
x-amazon-apigateway-request-validators:
  $ref: ./gateway/validators.yml
x-amazon-apigateway-gateway-responses:
  $ref: ./gateway/responses.yml
definitions:
  $ref: ./definitions.yml
paths:
  /health:
    get:
      $ref: ./operations/getHealth.yml
  /products:
    get:
      $ref: ./operations/getProducts/getProducts.yml
    post:
      $ref: ./operations/postProduct/postProduct.yml
  /products/{id}:
    get:
      $ref: ./operations/getProduct/getProduct.yml