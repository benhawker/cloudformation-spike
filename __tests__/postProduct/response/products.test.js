import api from 'api-gateway-template-tester';

const validator = global.loadSchema('src/definitions/empty.yml');
const template = global.loadVelocityTemplate(
  'src/operations/postProduct/response.yml', 'application/json');

import * as dynamoResponse from './mock/mock-response.json';

describe('Validating response templates', () => {
  let test;
  let renderedResponse;
  let validationResponse;
  let validationErrors;

  beforeEach(() => {
    console.log(dynamoResponse);
    
    test = api().body(dynamoResponse);
    renderedResponse = test.render(template);
    
    console.log(renderedResponse);
    validationResponse = validator(JSON.parse(renderedResponse));
    
    console.log(renderedResponse);

    if (!validationResponse) {
      validationErrors = validator.errors;
    }
  });

  describe('Rendered correctly', () => {
    it('Produces valid JSON', () => {
      console.log(renderedResponse);
      expect(renderedResponse).toMatchSnapshot();
    });

    it('Validated correctly', () => {
      console.log(validationErrors);
      expect(validationResponse).toEqual(true);
    });
  });
});
