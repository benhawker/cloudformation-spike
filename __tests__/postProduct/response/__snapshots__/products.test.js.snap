// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validating response templates Rendered correctly Produces valid JSON 1`] = `
"{}
"
`;
